apps/examples/wakerec/wakerec.cxx

```diff
@@ -327,34 +328,51 @@ extern "C" {
 int wakerec_main(int argc, char *argv[])
 {
 	printf("wakerec_main Entry\n");
-	sd = media::voice::SpeechDetector::instance();
-	if (!sd->initKeywordDetect(16000, 1)) {
-		printf("#### [SD] init failed.\n");
-		return 0;
+
+	if (argc > 3) {
+		printf("invalid input\n");
+		printf("Usage : wakerec [mode]\n");
+		printf("mode is optional 0 Disable wakeup\n");
+		return -1;
 	}
+	auto recorder = std::shared_ptr<WakeRec>(new WakeRec());
+	if (argc == 2 && atoi(argv[1]) == 0) {
+		printf("disable KD!!\n");
+		recorder->initWareRec(false);
+		recorder->startRecord();
+	} else {
+		recorder->initWareRec(true);
+		sd = media::voice::SpeechDetector::instance();
+		if (!sd->initKeywordDetect(16000, 1)) {
+			printf("#### [SD] init failed.\n");
+			return 0;
+		}
 
-	sd->addListener(std::make_shared<WakeRec>());
+		sd->addListener(recorder);
 
-	printf("###################################\n");
-	printf("#### Wait for wakeup triggered ####\n");
-	printf("###################################\n");
-	
-	if (sd->getKeywordBufferSize(&bufferSize) == true) {
-		printf("KD buffer size %d\n", bufferSize);
-		gBuffer = new uint8_t[bufferSize];
-		if (!gBuffer) {
-			printf("memory allocation failed\n");
+		printf("###################################\n");
+		printf("#### Wait for wakeup triggered ####\n");
+		printf("###################################\n");
+		
+		if (sd->getKeywordBufferSize(&bufferSize) == true) {
+			printf("KD buffer size %d\n", bufferSize);
+			gBuffer = new uint8_t[bufferSize];
+			if (!gBuffer) {
```

Comment:
if !NULL only work when used like new(std::nothrow)

Reply:
nullptr is \0 so I think this is okay.